NDTensors/src/lib/SetParameters/src/typeparameteraccessor.jl

```diff
@@ -0,0 +1,40 @@
+unionall_to_datatype(type::Type) = Base.unwrap_unionall(type)
+function datatype_to_unionall(type::Type, unionall_reference::Type)
+  return Base.rewrap_unionall(type, unionall_reference)
+end
+parameters(type::DataType) = Tuple(type.parameters)
+get_parameter(type, position::Int) = parameters(type)[position]
+unspecify_parameters(type::DataType) = Base.typename(type).wrapper
```

Comment:
Let's leave a comment that this code relies on Base Julia internals.

Reply:
Okay I added that comment!